### ISSUE TITLE
Can't package project - llama.h not found

### ISSUE THREAD

**Issue by aleistor222** [mika314/UELlama#8]
I'm having issues trying to package my project on Unreal Engine 5.3 when compiling this plugin.
The plugin works fine in the editor but while packaging I get the error message:

```
UATHelper: Packaging (Android (ASTC)): In file included from ../../Projects/MixedReality/Intermediate/Build/Android/a/MixedReality/Development/UELlama/Module.UELlama.cpp:4:
UATHelper: Packaging (Android (ASTC)): C:/Unreal Engine 5/UnrealEngine/Projects/MixedReality/Plugins/UELlama/Source/UELlama/Private/LlamaComponent.cpp(18,10): fatal error: 'llama.h' file not found
UATHelper: Packaging (Android (ASTC)): #include "llama.h"
UATHelper: Packaging (Android (ASTC)):          ^~~~~~~~~
UATHelper: Packaging (Android (ASTC)): 1 error generated.
```

Has anyone else encountered this issue and know a solution?

**Comment by aleistor222** [comment]
I've tried including the llama.h file from llama.cpp project, this then requires more and more files from llama.cpp and then when all the relevant files are included I end up with errors like:
C:\Unreal Engine 5\UnrealEngine\Projects\MixedReality\Plugins\UELlama\Source\UELlama\Private\LlamaComponent.cpp(33): error C2660: 'llama_token_to_piece': function does not take 4 arguments
  C:\Unreal Engine 5\UnrealEngine\Projects\MixedReality\Plugins\UELlama\Source\UELlama\Private\LlamaComponent.cpp(30): note: see declaration of 'llama_token_to_piece'
  C:\Unreal Engine 5\UnrealEngine\Projects\MixedReality\Plugins\UELlama\Source\UELlama\Private\LlamaComponent.cpp(33): note: while trying to match the argument list '(const llama_context *, llama_token, _Ty *, unsigned __int64)'
          with
          [
              _Ty=char
          ]

Is it possible to get any assistance on this please?

**Comment by mika314** [comment]
Make sure you are using the source code of `lama.cpp` from the following hash `a40f2b656fab364ce0aff98dbefe9bd9c3721cc9`

**Comment by aleistor222** [comment]
Sorry to keep bothering you on this, I'm not sure I understand.
Am I meant to be copying files over from that ggerganov/llama.cpp project with the correct hash? I see there are the necessary llama.h and other files already in the includes folder of this project. So I'm confused where llama.cpp comes into it?

**Comment by aleistor222** [comment]
I'm running on Windows and when I try to follow the instructions to build using cmake I get the error CMakeLists.txt doesn't exist.
So I build through the Visual Studio IDE.
Is there a different way to do this?

**Comment by mika314** [comment]
No, copying is not necessary. I'm not sure why you are getting that error; it does look like a header version mismatch. I never built the plugin on Windows, but I know some people were able to successfully build it.

**Comment by matiasgql** [comment]
You are using a new version headers of llama.cpp you will need to use the ones of the release b1258 with the hash mentioned above. The reason is that I copied llama_token_to_piece into the code and now it is inside llama.h, so when you try to compile with new headers, the build doesn't know what llama_token_to_piece should take.
